R/install.R

```diff
@@ -1,49 +1,68 @@
 #' Installs python dependencies
-#' @param python_version The version of python to install if not available
-#' @param virtualenv_name The name of the Virtual Environment to use to
-#' install the python libraries. Defaults to "r-sparklyr".
-#' @param force Flag that tells `reticulate` to re-create the Python Virtual
-#' Environment even if one with the same name already exists
-#' @param ignored_installed Flag that tells `reticulate` to ignore the Python
-#' library installation if the given library is already installed
-#' @param method Installation method. By default, "auto" automatically finds
-#'  a method that will work in the local environment. Change the default to
-#'  force a specific installation method. Note that the "virtualenv" method is
-#'   not available on Windows.
+#' @param envname The name of the Python Environment to use to install the
+#'   Python libraries. Defaults to "r-sparklyr".
+#' @param python_version The version of Python to use to create the Python
+#'   environment.
+#' @param new_env If `TRUE`, any existing Python virtual environment and/or
+#'   Conda environment specified by `envname` is deleted first.
+#' @param method The installation method to use. If creating a new environment,
+#'   `"auto"` (the default) is equivalent to `"virtualenv"`. Otherwise `"auto"`
+#'   infers the installation method based on the type of Python environment
+#'   specified by `envname`.
+#' @param ... Passed on to [`reticulate::py_install()`]
 #' @export
-install_pyspark <- function(virtualenv_name = "r-sparklyr",
-                            python_version = NULL,
-                            force = FALSE,
-                            ignored_installed = TRUE,
-                            method = c("auto", "virtualenv", "conda")
-                            ) {
-
-  version <- ">=3.9"
-
-  pkgs <- c(
-    "pyspark", "pandas", "PyArrow", "grpcio", "google-api-python-client",
-    "grpcio_status", "databricks-connect", "delta-spark"
+install_pyspark <- function(envname = "r-sparklyr",
+                            ...,
+                            python_version = ">=3.9",
+                            new_env = identical(envname, "r-sparklyr"),
```

Comment:
The thinking is that users will only call this function if something went wrong. E.g., they messed up the environment by installing conflicting dependencies, and now any interaction with `pip` errors. The hope is that a user calling this function is like Dorthy clicking her heels three times--it instantly brings you to a place where everything is right.

Reply:
Oh cool! In that case, should we just make the default to `TRUE`, so that way it will behave like that even if the  user inputs an `envname` not "r-sparklyr"?

Reply:
I don't think so, we should only aggressively delete environments that we know are "ours". Also, I think it's sensible for users to install pyspark into other environments. E.g., if a user wants to install tensorflow and pyspark in the same environment, this should just work:
```r
tensorflow::install_tensorflow("./venv")
pysparklyr::install_pyspark("./venv")
```

Reply:
Oh I see, that makes sense